ui/src/components/forms/UserFormContainer.js

```diff
@@ -23,6 +24,38 @@ const UserFormContainer = (props) => {
         // Form is not published, hence it is unavailable
         if (!data[0]) history.push('/form-unavailable');
         else {
+          const formAvailableFrom = DateTime.fromISO(
+            data[0].formAvailableFrom
+          ).setZone('Asia/Hong_Kong');
+          const formAvailableUntil = DateTime.fromISO(
+            data[0].formAvailableUntil
+          ).setZone('Asia/Hong_Kong');
+
+          // One of these values are set
+          if (formAvailableFrom.isValid || formAvailableUntil.isValid) {
+            // If form available from is set & valid, check if current time is
+            // after formAvailableFrom
+            let afterStartTime = formAvailableFrom.isValid
+              ? DateTime.now().setZone('Asia/Hong_Kong') >= formAvailableFrom
+              : true;
+
+            // Compound checking: if form available until:
+            // is valid: check if it's within the time of form availability
+            // is not valid: default to whatever value was already there
+            let beforeEndTime = formAvailableUntil.isValid
+              ? DateTime.now().setZone('Asia/Hong_Kong') < formAvailableUntil
+              : true;
+            let isInRange = afterStartTime && beforeEndTime;
+
+            if (!isInRange) {
+              if (!beforeEndTime) history.push('/form-unavailable');
+              else
+                history.push({
+                  pathname: '/form-will-open',
```

Comment:
hmmm. have a feeling people will complain about this. since we are redirecting to an error page, refreshing the page won't bring them back to this form. i personally sometimes go to a form at like 7:59 and refresh until it opens up